### PR TITLE
Fix scalar mask to bool

### PR DESCRIPTION
**Pull request by ronlieb**
Received this problem from one of our customers.

### CODE REVIEW

File: test/correctness/mask_to_bool_scalar.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "Halide.h"
+
+using namespace Halide;
+
+int main() {
```
Comment (zvookin) [inline test/correctness/mask_to_bool_scalar.cpp]: Per the Travis failure, this test needs to check if Hexagon is enabled in the target and if not, print a warning and exit with a success (0) indication.

**Comment by dsharletg** [comment]
Did you intend to fork this half way through my other branch? Are the changes to HexagonOptimize actually necessary?

File: src/CodeGen_Hexagon.cpp [hunk 2]
```diff
@@ -1517,8 +1517,17 @@ void CodeGen_Hexagon::visit(const Call *op) {
         } else if (op->is_intrinsic(Call::cast_mask)) {
             internal_error << "cast_mask should already have been handled in HexagonOptimize\n";
         }
+    } else if (op->is_intrinsic(Call::bool_to_mask)) {
```
Comment (dsharletg) [inline src/CodeGen_Hexagon.cpp]: This code already exists above in the if vector branch. How about just moving that if outside the branch, and not as an else condition?

At least, the old code should be simplified assuming the if (vector) condition is true, due to the outer if.

**Comment by ronlieb** [comment]
Abandon this branch, pushing a new one

**Pull request closed without merging**